test/apps/iodemo/io_demo.cc

```diff
@@ -2916,10 +2953,19 @@ static int parse_args(int argc, char **argv, options_t *test_opts)
     }
 
     while (optind < argc) {
-        test_opts->servers.push_back(argv[optind++]);
+        servers.push_back(argv[optind++]);
     }
 
-    adjust_opts(test_opts);
+    if (!test_opts->src_addrs.empty() &&
+        (test_opts->src_addrs.size() != 1) &&
+        (test_opts->src_addrs.size() != servers.size())) {
+        std::cout << "In client size, source addr number should be 1 "
+                  << "OR equal server addr number" << std::endl;
+        return -1;
+    }
+
```

Comment:
why is this needed?

Reply:
It requires the src_addrs are used to connect with the server one by one
1) ```empty``` case: use default destination reachable src ip to connect with server.
2) ```1``` case: the only given src ip could connect with all servers.
3) ```src_addrs.size() = servers.size()``` case: the src addresses are used one by one to connect with the servers.

Reply:
Removed the check here after simplifying the design